src/mongocxx/options/index.hpp

```diff
@@ -0,0 +1,365 @@
+    const stdx::optional<bsoncxx::document::view>& weights() const;
+
+    ///
+    /// For text indexes, the language that determines the list of stop words
+    /// and the rules for the stemmer and tokenizer.
+    ///
+    /// @param default_language
+    ///   The default language for text indexes.
+    ///
+    void default_language(std::string default_language);
+
+    ///
+    /// The current default_language setting.
+    ///
+    /// @return The current default_language.
+    ///
+    const stdx::optional<std::string>& default_language() const;
+
+    ///
+    /// For text indexes, the name of the field, in the collection’s documents,
+    /// that contains the override language for the document.
+    ///
+    /// @param language_override
+    ///   The name of the field that contains the override language for text indexes.
+    ///
+    void language_override(std::string language_override);
+
+    ///
+    /// The current language_override setting.
+    ///
+    /// @return The current language_override.
+    ///
+    const stdx::optional<std::string>& language_override() const;
+
+    ///
+    /// Sets the document for the partial filter expression for partial
+    /// indexes.
+    ///
+    /// @param partial_filter_expression
+    ///   The partial filter expression document.
+    ///
+    void partial_filter_expression(bsoncxx::document::view partial_filter_expression);
+
+    ///
+    /// The current partial_filter_expression setting.
+    ///
+    /// @return The current partial_filter_expression.
+    ///
+    const stdx::optional<bsoncxx::document::view>& partial_filter_expression() const;
+
+    ///
+    /// For 2dsphere indexes, the 2dsphere index version number. Version can be
+    /// either 1 or 2.
+    ///
+    /// @param twod_sphere_version
+    ///   The 2dsphere index version number.
+    ///
+    void twod_sphere_version(std::uint8_t twod_sphere_version);
+
+    ///
+    /// The current twod_sphere_version setting.
+    ///
+    /// @return The current twod_sphere_version.
+    ///
+    const stdx::optional<std::uint8_t>& twod_sphere_version() const;
+
+    ///
+    /// For 2d indexes, the number of precision of the stored geohash value of
+    /// the location data.
+    ///
+    /// @param twod_bits_precision
+    ///   The number of precision of the stored geohash value.
+    ///
+    void twod_bits_precision(std::uint8_t twod_bits_precision);
+
+    ///
+    /// The current twod_bits_precision setting.
+    ///
+    /// @return The current twod_bits_precision.
+    ///
+    const stdx::optional<std::uint8_t>& twod_bits_precision() const;
+
+    ///
+    /// For 2d indexes, the lower inclusive boundary for the longitude and
+    /// latitude values.
+    ///
+    /// @param twod_location_min
+    ///   The lower inclusive boundary.
+    ///
+    void twod_location_min(double twod_location_min);
+
+    ///
+    /// The current twod_location_min setting.
+    ///
+    /// @return The current twod_location_min.
```

Comment:
take your more descriptive comment from above, the current lower inclusive boundary for longitude and latitude values.